### PR TITLE
RELATED: RAIL 2041 Make Execute component more convenient to use

### PR DESCRIPTION
**Pull request by kandl**
- Add support for error & loading components
- Update examples

JIRA: RAIL-2041

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3348/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5000/ : FAILURE in 5m 49s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3349/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5001/ : SUCCESS in 12m 15s

File: libs/sdk-ui/src/execution/RawExecute.tsx [hunk 1]
```diff
@@ -48,12 +49,39 @@ export interface IRawExecuteProps extends IWithLoadingEvents<IRawExecuteProps> {
...
+    if (ErrorComponent && error) {
+        return <ErrorComponent error={error} />;
+    }
+
+    if (LoadingComponent && isLoading) {
+        return <LoadingComponent />;
+    }
+
+    if (LoadingComponent && ErrorComponent && !result) {
+        return null;
```
Comment (no23reason) [inline libs/sdk-ui/src/execution/RawExecute.tsx:83]: What is the reasoning behind this? Why not just call the children with the nullish result? What if I want ot provide Error and Loading but want to handle the empty result myself? Or that will never happen?

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1290

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1003/ : SUCCESS in 56s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1273/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5008/ : SUCCESS in 12m 13s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/401/ : SUCCESS in 9m 09s (non-voting)

**Pull request merged**